### PR TITLE
port of pr 10153 into beta.

### LINKED ISSUE #5337: Windows Explorer crashes by going to the save-button  of metadata
**Reported by The_Dark_Man on 2015-09-09 15:08**
When I am in the Windows Explorer on a file with metadata and push the tab key I go through the Metadata Interpret, genre and so on. After this, there is “save” and “cancel”. When I tab on “save”, the windows explorer crashes. It’s closed. This bug is in NVDA 2015.3 new. I think, this isn’t correct. I had this problem with a mp3-file, for example. I use Windows 7 Professional. I tried it with NVDA 2015.2 ant there was no problem.

### PR DESCRIPTION
**Pull request by lukaszgo1**
### Link to issue number:
Fixes #5337 port of #10153 into beta.

### Testing performed:
Ensured that the crash no longer occurs.
### Known issues with pull request:
None
### Change log entry:
Same as in #5337

### CODE REVIEW

**Comment by Brian1Gaff** [comment]
Was there not a second fix for  a problem with the mouse being used in 
certain circumstances? This was the most annoying one we had, since sighted 
rather than turn off nvda, merely muted the sound then had nearly 50 
explorer crashes in 10 minutes.

Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

File: source/appModules/explorer.py [hunk 1]
```diff
@@ -189,6 +191,19 @@ def _get_windowText(self):
...
+class MetadataEditField(RichEdit50):
+	""" Used for metadata edit fields in Windows Explorer in Windows 7.
+	By default these fields would use ITextDocumentTextInfo ,
+	but to avoid Windows Explorer crashes we need to use EditTextInfo here. """
+	@classmethod
+	def _get_TextInfo(cls):
+		if ((winVersion.winVersion.major, winVersion.winVersion.minor) == (6, 1)):
+			cls.TextInfo = EditTextInfo
+		else:
+			cls.TextInfo = super().TextInfo
```
Comment (leonardder) [inline source/appModules/explorer.py]: This is not compatible with Python 2

```suggestion
			cls.TextInfo = super(MetadataEditField, self).TextInfo
```

**Review by LeonarddeR — CHANGES_REQUESTED** [review]

**Comment by lukaszgo1** [comment]
@leonardder All done

File: source/appModules/explorer.py [hunk 2]
```diff
@@ -189,6 +191,19 @@ def _get_windowText(self):
...
+class MetadataEditField(RichEdit50):
+	""" Used for metadata edit fields in Windows Explorer in Windows 7.
+	By default these fields would use ITextDocumentTextInfo ,
+	but to avoid Windows Explorer crashes we need to use EditTextInfo here. """
+	@classmethod
+	def _get_TextInfo(cls):
+		if ((winVersion.winVersion.major, winVersion.winVersion.minor) == (6, 1)):
+			cls.TextInfo = EditTextInfo
+		else:
+			cls.TextInfo = super(MetadataEditField, self).TextInfo
```
Comment (leonardder) [inline source/appModules/explorer.py]: Ugh. I made this mistake in my initial comment and updated the comment afterwards. Yet, the initial version of the comment ended up in the code:

```suggestion
			cls.TextInfo = super(MetadataEditField, cls).TextInfo
```

**Comment by michaelDCurran** [comment]
> 
> Was there not a second fix for a problem with the mouse being used in certain circumstances? This was the most annoying one we had, since sighted rather than turn off nvda, merely muted the sound then had nearly 50 explorer crashes in 10 minutes.

Yes, this is already in the beta branch.

**Comment by michaelDCurran** [comment]
@leonardder I think you have a review pending on this pr.

**Review by LeonarddeR — APPROVED** [review]
sorry for missing this @michaelDCurran

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**